bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/internal/discovery/CBusGroupDiscovery.java

```diff
@@ -0,0 +1,114 @@
+            ThingUID bridgeUid = cbusNetworkHandler.getThing().getBridgeUID();
+            if (bridgeUid == null) {
+                scanFinished();
+                return;
+            }
+            try {
+                Map<Integer, ThingTypeUID> applications = new HashMap<Integer, ThingTypeUID>();
+                applications.put(CBusBindingConstants.CBUS_APPLICATION_LIGHTING, CBusBindingConstants.THING_TYPE_LIGHT);
+                applications.put(CBusBindingConstants.CBUS_APPLICATION_DALI, CBusBindingConstants.THING_TYPE_DALI);
+                applications.put(CBusBindingConstants.CBUS_APPLICATION_TEMPERATURE,
+                        CBusBindingConstants.THING_TYPE_TEMPERATURE);
+                applications.put(CBusBindingConstants.CBUS_APPLICATION_TRIGGER,
+                        CBusBindingConstants.THING_TYPE_TRIGGER);
+
+                Network network = cbusNetworkHandler.getNetwork();
+                if (network == null) {
+                    scanFinished();
+                    return;
+                }
+                for (Map.Entry<Integer, ThingTypeUID> applicationItem : applications.entrySet()) {
+                    Application application = network.getApplication(applicationItem.getKey());
+                    if (application == null) {
+                        continue;
+                    }
+                    ArrayList<Group> groups = application.getGroups(false);
+                    for (Group group : groups) {
+                        logger.debug("Found group: {} {} {}", application.getName(), group.getGroupID(),
+                                group.getName());
+                        Map<String, Object> properties = new HashMap<>();
+                        properties.put(CBusBindingConstants.PROPERTY_APPLICATION_ID,
+                                Integer.toString(applicationItem.getKey()));
+                        properties.put(CBusBindingConstants.CONFIG_GROUP_ID, Integer.toString(group.getGroupID()));
+                        properties.put(CBusBindingConstants.PROPERTY_GROUP_NAME, group.getName());
+                        properties.put(CBusBindingConstants.PROPERTY_NETWORK_ID,
+                                Integer.toString(network.getNetworkID()));
+
+                        ThingUID uid = new ThingUID(applicationItem.getValue(), Integer.toString(group.getGroupID()),
+                                bridgeUid.getId(), cbusNetworkHandler.getThing().getUID().getId());
+                        DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                                .withLabel("CBUS " + group.getName() + "(" + group.getGroupID() + ")")
+                                .withBridge(cbusNetworkHandler.getThing().getUID()).build();
+                        thingDiscovered(result);
+                    }
+                }
+            } catch (CGateException e) {
+                logger.warn("Failed to discover groups", e);
```

Comment:
Should we log the exception on `WARN` level? Or maybe only on debug?

Reply:
Changed to debug